packages/builder/src/components/common/TemplateCard.svelte

```diff
@@ -0,0 +1,123 @@
+<script>
+  export let backgroundColour
+  export let imageSrc
+  export let name
+  export let icon
+  export let overlayEnabled = true
+
+  let imageError = false
+  let imageLoaded = false
+
+  const imageRenderError = () => {
+    imageError = true
+  }
+
+  const imageLoadSuccess = () => {
+    imageLoaded = true
+  }
+</script>
+
+<div class="template-card" style="background-color:{backgroundColour};">
+  <div class="template-thumbnail card-body">
+    <img
+      alt={name}
+      src={imageSrc}
+      on:error={imageRenderError}
+      on:load={imageLoadSuccess}
+      class={`${imageLoaded ? "loaded" : ""}`}
+    />
+    <div style={`display:${imageError ? "block" : "none"}`}>
+      <svg
+        width="26px"
+        height="26px"
+        class="spectrum-Icon"
+        style="color: white"
+        focusable="false"
+      >
+        <use xlink:href="#spectrum-icon-18-{icon}" />
+      </svg>
+    </div>
+    <div class={overlayEnabled ? "template-thumbnail-action-overlay" : ""}>
```

Comment:
I love these cards! One improvement I think we could make would be to slightly transition the appearance of the overlay to be in line with the rest of the spectrum feel.

I would recomment:
- Set overlay opacity to 0, and leave display as flex always
- On hover, set the overlay opacity to 1
- Add `transition: opacity var(--spectrum-global-animation-duration-100) ease` to match how spectrum components do it

Reply:
Sound nice, I'll give it a go!